dispatcher/backend/src/routes/utils.py

```diff
@@ -97,14 +97,16 @@ def remove_s3_secrets(response: dict):
         else:
             if not isinstance(response[key], str) or "://" not in response[key]:
                 continue
-            url = urlparse(response[key])
-            response[key] = url._replace(
-                query="&".join(
-                    [
-                        f"{key}={value}"
-                        for key, values in parse_qs(url.query).items()
-                        if str(key).lower() not in ["keyid", "secretaccesskey"]
-                        for value in values
-                    ]
-                )
-            ).geturl()
+            for part in [part for part in response[key].split() if "://" in part]:
+                url = urlparse(part)
+                url = url._replace(
+                    query="&".join(
```

Comment:
Again, why not using `urlencode(doseq=True)` to reconstruct?

Reply:
You deserialize with `parse_qs` so you should serialize back with `urlencode` as [the doc recommends](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs). I know the function name is misleading…

> Use the [urllib.parse.urlencode()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) function (with the doseq parameter set to True) to convert such dictionaries into query strings.